src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -136,6 +140,8 @@ public bool TryRegister(SafeSocketHandle socket, out Interop.Error error)
         //
         private bool IsFull { get { return _nextHandle == MaxHandles; } }
 
+        private bool IsAio => _aioContext.Ring != null;
```

Comment:
this might need a cleaner check or even moving entire AIO logic to a standalone type